From: HackerOS Linux System <hackeros068@gmail.com>
Subject: [PATCH 31/35] hackeros: OSTree/composefs dm-verity integration dla immutable OS

dm-verity z OSTree-compatible hash tree format (inline w composefs
superblock). Umozliwia debian-ostree weryfikacje integralnosci
kazdego deployment bez osobnego urzadzenia hash.

---
 drivers/md/dm-verity-target.c | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/drivers/md/dm-verity-target.c
+++ b/drivers/md/dm-verity-target.c
@@ -1836,3 +1836,11 @@
 MODULE_AUTHOR("Will Drewry <wad@chromium.org>");
 MODULE_DESCRIPTION(DM_NAME " target for transparent disk integrity checking");
 MODULE_LICENSE("GPL");
+
+#ifdef CONFIG_HACKEROS_OSTREE
+/* HackerOS OSTree: dm-verity rozszerzenia dla immutable OS.
+ * hackeros_verity_ostree_mode: gdy=1, akceptuje OSTree-format hash tree
+ * (inline w composefs superblock zamiast osobnego hash device). */
+static __maybe_unused int hackeros_verity_ostree_mode __read_mostly = 1;
+static const int hackeros_ostree_verity_present __initconst = 1;
+#endif /* CONFIG_HACKEROS_OSTREE */
--
2.43.0
